src/MainMenu.cpp

```diff
@@ -113,3 +117,172 @@ ZunResult MainMenu::LoadTitleAnm(MainMenu *menu)
     return ZUN_SUCCESS;
 }
 #pragma optimize("", on)
+
+#pragma optimize("s", on)
+#pragma var_order(i, drawVm)
+ZunResult MainMenu::DrawStartMenu(void)
+{
+    int i;
+    i = MoveCursor(this, 8);
+    if ((this->cursor == 1) && !g_GameManager.hasReachedMaxClears(0, 0) && !g_GameManager.hasReachedMaxClears(0, 1) &&
+        !g_GameManager.hasReachedMaxClears(1, 0) && !g_GameManager.hasReachedMaxClears(1, 1))
+    {
+        this->cursor += i;
+    }
+    AnmVm *drawVm = this->vm;
+    for (i = 0; i < 8; i++, drawVm++ /* zun why */)
+    {
+        DrawMenuItem(drawVm, i, this->cursor, 0xffff0000, 0x80300000, 122);
+    }
+    if (this->stateTimer >= 0x14)
+    {
+        if (WAS_PRESSED(0x1001))
```

Comment:
It'd be nice to have some defines for the key. This is likely `KEY_ENTER | KEY_SHOOT`.

We have a comment in GetInput:

```
    GetKeyboardState((PBYTE)keyboardState);
                    /* 38  = VK_UP = KEY_UP
                       40  = VK_DOWN = KEY_DOWN
                       37  = VK_LEFT = KEY_LEFT
                       39  = VK_RIGHT = KEY_RIGHT
                       104 = VK_NUMPAD8 = KEY_UP
                       98  = VK_NUMPAD2 = KEY_DOWN
                       100 = VK_NUMPAD4 = KEY_LEFT
                       102 = VK_NUMPAD6 = KEY_RIGHT
                       103 = VK_NUMPAD7 = KEY_LEFT | KEY_UP
                       105 = VK_NUMPAD9 = KEY_RIGHT | KEY_UP
                       97  = VK_NUMPAD1 = KEY_LEFT | KEY_DOWN
                       99  = VK_NUMPAD3 = KEY_RIGHT | KEY_DOWN
                       36  = VK_HOME = KEY_HOME
                       90  = Z = KEY_BOMB
                       88  = X = KEY_BOMB
                       16  = VK_SHIFT = KEY_FOCUS
                       27  = VK_ESCAPE = KEY_MENU
                       17  = VK_CONTROL = KEY_SKIP
                       81  = Q = KEY_Q
                       83  = S = KEY_S
                       13  = VK_RETURN = KEY_ENTER */
                    /* This sets a bitfield of which buttons are pressed.
                       
                       0x001 is shoot
                       0x002 is BOMB
                       0x004 is FOCUS
                       0x008 is MENU
                       0x010 is UP
                       0x020 is DOWN
                       0x040 is LEFT
                       0x080 is RIGHT
                       0x100 is SKIP
                       0x200 is KEY_Q
                       0x400 is KEY_S
                       0x1000 is KEY_ENTER */
```

Reply:
Ohhh, I didn't know about those, looks like I haven't familiarised myself with the codebase as much as I thought

Reply:
To be fair, I literally just finished figuring those out :P